planetiler-core/src/test/java/com/onthegomap/planetiler/util/WikidataTest.java

```diff
@@ -161,6 +171,43 @@ void testRetryFailedRequestOnce() throws IOException, InterruptedException {
     assertInstanceOf(IOException.class, innerException);
   }
 
+  @Test
+  void testLegacyWikidataNamesJson() throws IOException {
+    String json = wikidataNamesJson.replaceAll(",\"<timestamp .>\"", "");
+    var reader = new BufferedReader(new StringReader(json));
+    // no timestamp + age limit set => all old => all should be dropped
+    var translationsProvider = Wikidata.load(reader, Duration.ofSeconds(1), 0);
+    assertEquals(0, translationsProvider.getAll().size());
+  }
+
+  @Test
+  void testWikidataNamesJsonMaxAge() throws IOException {
+    Duration maxAge = Duration.ofSeconds(1);
+    Instant fresh = Instant.now();
+    Instant old = fresh.minus(maxAge).minus(maxAge);
+
+    String json = wikidataNamesJson
+      .replaceAll("<timestamp 1>", fresh.toString())
+      .replaceAll("<timestamp .>", old.toString());
+
+    var reader = new BufferedReader(new StringReader(json));
+    var translationsProvider = Wikidata.load(reader, maxAge, 0);
+    assertEquals(1, translationsProvider.getAll().size());
+  }
+
+  @Test
+  void testWikidataNamesJsonUpdateLimit() throws IOException {
+    Duration maxAge = Duration.ofSeconds(1);
+    Instant old = Instant.now().minus(maxAge).minus(maxAge);
```

Comment:
Let's not rely on current time in tests, instead can we make a test `Clock` at a fixed time and pass it into `static WikidataTranslations load(BufferedReader reader, ...` method?

Reply:
Adjusted.